Calculate audio energy and send in RTP header extension

juberti opened the pull request:
Fixes #614

juberti commented:
@jlaine

jlaine commented:
Hi @juberti thanks for the PR!

Some notes:

- I've let CI run and it looks like the linters are not happy
- maybe it would make sense to pull the `_compute_level_dbov` method out to a plain function (for instance in `aiortc.rtp`) so that we can have a unit test with some test vectors?
- I see `level` is set to `None` for video frames, isn't this going to blow up: `packet.extensions.audio_level = (False, -enc_frame.level)`
- how costly is this calculation?

jlaine commented on src/aiortc/rtcrtpsender.py:
```diff
@@ -36,12 +38,19 @@
+    def __init__(self, payloads: List[bytes], timestamp: int, level: int):
+        self.payloads = payloads
+        self.timestamp = timestamp
+        self.level = level
```
Maybe `audio_level` would be more explicit?

juberti commented:
OK, fixed the lint issues and the video-frame issue, and broke out the energy calc function and added test for it. 

As for costs, it's small but not nothing, maybe 3% of a core on my system. numpy might be an option if this is too much, although there might be other costs incurred roundtripping to numpy and back.

jlaine commented:
I ran `isort` on the code which fixes additional linter errors.

juberti commented:
Thanks for doing that, I had run flake8 and black but not isort.

jlaine commented:
For my information, does this header need to be negotiated or are we good to always send it?

juberti commented:
It should be negotiated, that's right. I think there is already logic to only send the header if it's been negotiated, although it will always be computed.

jlaine commented:
OK, I'm going to merge this PR as is, even though I'm a bit concerned by the computational overhead (some of aiortc's usecase are on low power devices).

If issues suddenly get reported related to performance, would you consider working on a follow up PR to make this extension optional? I noticed that webrtc-extensions mentions a `RTCRtpTransceiver.setOfferedRtpHeaderExtensions` method, maybe that could be used?

jlaine commented:
Oops, I forgot to squash the commits, so I've had to force-push to `main`..

juberti commented:
Will you be publishing a new rev of aiortc with these changes?
